af/fastpath/fastpath/core.py

```diff
@@ -876,17 +876,28 @@ def score_vanilla_tor(msm):
     if msm["software_name"] == "ooniprobe" and all_keys_none(tk, nks):
         if tk["tor_progress"] == 0:
             # client bug?
+            scores["accuracy"] = 0.0
             scores["msg"] = "Client bug"
             return scores
 
-    tor_log = tk.get("tor_log", None)
-    if tor_log is None:
+    tor_logs = []
+    # Handle incompatibility with legacy test:
+    # https://github.com/ooni/spec/blob/master/nettests/ts-016-vanilla-tor.md#incompatibility-with-ooniprobe-legacy
+    if "tor_logs" in tk and tk["tor_logs"]:
+        tor_logs = tk["tor_logs"]
+    elif "tor_log" in tk and tk["tor_log"]:
+        tor_logs = tk["tor_log"].split("\n")
+
+    if len(tor_logs) == 0:
         # unknown bug
+        scores["accuracy"] = 0.0
         return scores
 
-    if (
-        "Bootstrapped 100%: Done" in tor_log
-        or "Bootstrapped 100% (done): Done" in tor_log
+    if any(
```

Comment:
The [] should not be needed, also can you move the strings ouf of the any?

Reply:
>The [] should not be needed

ah yes, good call!

> can you move the strings ouf of the any

I'm not sure I understand this suggestion? Should I define the strings as a list and do two nested `any`s or just define the strings as constants and then refer to the constant inside of the current `any` (or something else)?

Reply:
(@hellais is on holidays. Implemented in a new commit)